### PR TITLE
feat: Support shell expansion for file arguments

### LINKED ISSUE #39: Expand shell expressions for files
**Is your feature request related to a problem? Please describe.**

Shell expressions are not expanded for input/output files.

```
$ menyoki edit "~/test.png"
[2023-05-10T19:50:30 ERROR menyoki] IO error: `No such file or directory (os error 2)`
```

**Describe the solution you'd like**

Use a crate such as [`shellexpand`](https://gitlab.com/ijackson/rust-shellexpand) to handle these cases.

e.g. `~/test.png` should be expanded as `/home/orhun/test.png`

**Describe alternatives you've considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by JerelJr**
## Description

Added file expansion for commands that take file arguments. This was generally done the same way between the file-handling portions of the settings for each command. I found the sections where a command took file or directory input and instead of converting the next argument directly into a `PathBuf`, I first ran it through the `shellexpand::full` function. Something along the lines of:
```rust
let file = matches.value_of("file").unwrap_or_default();
let file = full(file)
		.map(|s| s.to_string())
		.unwrap_or(file.to_string());
let mut path = PathBuf::from(file);
```
This would expand input such as `"~/Pictures"` to `"/home/JerelJr/Pictures"` for example.
## Motivation and Context

closes #39
## How Has This Been Tested?

Ran `cargo test`. I was not sure how to add new tests for this as it would require opening some arbitrary file location under a user's home directory to run commands on. If there is something I am missing here feedback would be appreciated. 
## Types of changes

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🌱 New feature (non-breaking change which adds functionality)
- [ ] 🌵 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 📚 Documentation (no code change)
- [ ] 🧹 Refactor (refactoring production code)
- [ ] ⚙️ Other 

## Checklist:

- [x] 🎨 My code follows the code style of this project.
- [ ] 📚 I have updated the _documentation_ accordingly.
  - [ ] I have updated the [README.md](https://github.com/orhun/menyoki/blob/master/README.md)
  - [ ] I have updated the [CHANGELOG.md](https://github.com/orhun/menyoki/blob/master/CHANGELOG.md)
  - [ ] I have updated the [IMPLEMENTATION.md](https://github.com/orhun/menyoki/blob/master/IMPLEMENTATION.md)
  - [ ] I have updated the [configuration file](https://github.com/orhun/menyoki/blob/master/config/menyoki.conf).
  - [ ] I have updated the [man pages](https://github.com/orhun/menyoki/tree/master/man).
- [x] 📐 I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] 📊 I have added tests to cover my changes.
- [x] 📈 All new and existing tests passed.
- [x] 📎 [Clippy](https://github.com/rust-lang/rust-clippy) is not complaining.

### CODE REVIEW

**Comment by orhun** [comment]
Looks good overall! One thing that I think we should change is:

```diff
- let file = full(file)
+ let file = shellexpand::full(file)
```

I think having this as a explicit call is better since `full` can pretty much refer to anything else in another context. One alternative is importing it as an alias (e.g. `use shellexpand::full as expand_path`) but we can simply use it like above.

Regarding testing, you can edit one of the argument parsing tests and simply assert the result from the `from_parser` function. For example, you can use `~/` as a parameter and assert that it is the `dirs::home_dir` after expanded. Let me know if you need help about it!

**Comment by JerelJr** [comment]
Ah, okay. That makes sense. Will do.

**Comment by JerelJr** [comment]
I put the new test inside the `test_split_settings` function since the split command takes a `--dir` argument. Let me know if there are any other changes I need to make.

File: Cargo.toml [hunk 1]
```diff
@@ -70,6 +71,7 @@ features = ["suggestions", "color", "wrap_help"]
 
 [dev-dependencies]
 pretty_assertions = "1.3.0"
+dirs = "5.0.1"
```
Comment (orhun) [inline Cargo.toml:74]: I think we already have `dirs` in `[dependencies]` table so there is no need to specify it here.

File: src/anim/settings.rs [hunk 2]
```diff
@@ -2,6 +2,7 @@ use crate::args::matches::ArgMatches;
 use crate::args::parser::ArgParser;
 use crate::file::format::FileFormat;
 use crate::file::File;
+use shellexpand;
```
Comment (orhun) [inline src/anim/settings.rs:5]: We can simply call `shellexpand::full` without this import.

**Comment by orhun** [comment]
Looks good! Just 2 small changes and we're ready to go 🚀

**Review by orhun — APPROVED** [review]
Thanks for your contribution! 🐻

**Pull request merged**